src/ophyd_async/core/_signal.py

```diff
@@ -588,16 +588,15 @@ async def _wait_for_value():
                     break
 
         try:
-            await asyncio.wait_for(_wait_for_value(), timeout)
             if wait_for_set_completion:
+                await asyncio.wait_for(_wait_for_value(), timeout)
```

Comment:
I think this was correct as it was before, we always wait for `_wait_for_value` to complete, but we only `await status` if `wait_for_set_completion`